src/CTA.Rules.Update/CodeReplacer.cs

```diff
@@ -161,15 +163,24 @@ private List<GenericActionExecution> AddActionsWithoutExecutions(FileActions cur
 
         public List<GenericActionExecution> ValidateActions(List<GenericActionExecution> actions, string fileResult)
         {
-            var trimmedResult = Utils.EscapeAllWhitespace(fileResult);
+            // Matches all types of comments and strings
+            string regComments = @"\/\*(?:(?!\*\/)(?:.|[\r\n]+))*\*\/|\/\/(.*?)\r?\n|""((\\[^\n]|[^""\n])*)""|@(""[^""""]*"")+";
```

Comment:
Does this work for other types of comments? Example:
`// This is a comment`
or
`/// <summary>`
`/// Initializes a new instance of Solution Port, analyzing the solution path using the provided config.`
`/// </summary>`

I tried using a regex testing side and it doesn't seem to recognize them: http://regexstorm.net/tester

Can we load a syntax tree without comments? One example I was able to run was creating an extension method that loads the children without their trivia. This will ensure that we are using Roslyn APIs to get rid of comments instead of recreating our own:


        public static SyntaxNode NoComments(this SyntaxNode node)
        {
            if (node != null)
            {
                var childNodes = node.ChildNodes();
                node = node.ReplaceNodes(childNodes, RemoveComments);
                node = node.WithoutTrivia();
            }
            return node;
        }
        static SyntaxNode RemoveComments(SyntaxNode syntaxNode, SyntaxNode returnNode)
        {
            return syntaxNode.NoComments();
        }

Reply:
Yup it works on all types of comments, I used : https://regexr.com/ but your site also worked.
Please use:
\/\*(?:(?!\*\/)(?:.|[\r\n]+))*\*\/|\/\/(.*?)\r?\n|"((\\[^\n]|[^""\n])*)"|@("[^"]*")+
For the pattern and you can use any variety of comments to test against, I find the below input is pretty thorough:

// remove whole line comments
bool broken = false; // remove partial line comments
if (broken == true)
{
    return "BROKEN";
}

/// This is another comment

/* This // is a test/, or // This / is a test */
var comment="/* This is a test*/";
var url = "http://stackoverflow.com";
var abc = @" this /* \n is a comment in quote\n*/ with or without spaces between ";
var abc2 = "and /*or*/ and ";

/* remove block comments
else
{
    return "FIXED";
} // do not remove nested comments */ bool working = !broken;
return "NO COMMENT";

Reply:
I updated CR with the extensions all tests are still passing and comments are getting removed as intended. The only potential problem is if in the future we need to introduce more exceptions than just comments such as strings or others. I will keep the regex code commented out for potential re-introduction in the future if needed. I do think the extension idea is much cleaner however.